qcodes/dataset/data_set.py

```diff
@@ -1024,6 +1074,82 @@ def _load_to_dataframes(self, datadict: ParameterData) -> Dict[str, "pd.DataFram
             dfs[name] = self._data_to_dataframe(subdict, index)
         return dfs
 
+    def get_data_as_xarray(self,
+                           *params: Union[str,
+                                          ParamSpec,
+                                          _BaseParameter],
+                           concat: Optional[bool] = False,
+                           start: Optional[int] = None,
+                           end: Optional[int] = None) -> \
+            Dict[str, Union["xr.DataArray", "xr.Dataset"]]:
+        """
+        Returns the values stored in the :class:`.DataSet` for the specified parameters
+        and their dependencies as a dict of :py:class:`xr.DataArray`s
+        Each element in the dict is indexed by the names of the requested
+        parameters.
+
+        If no parameters are supplied data will be be
+        returned for all parameters in the :class:`.DataSet` that are not them self
+        dependencies of other parameters.
+
+        If provided, the start and end arguments select a range of results
+        by result count (index). If the range is empty - that is, if the end is
+        less than or equal to the start, or if start is after the current end
+        of the :class:`.DataSet` – then a dict of empty :py:class:`xr.DataArray`s is
+        returned.
+
+        Args:
+            *params: string parameter names, QCoDeS Parameter objects, and
+                ParamSpec objects. If no parameters are supplied data for
+                all parameters that are not a dependency of another
+                parameter will be returned.
+            concat: if True individual DataArrays are concatenated and returned
+                as a single xr.Dataset
+            start: start value of selection range (by result count); ignored
+                if None
+            end: end value of selection range (by results count); ignored if
+                None
+
+        Returns:
+            Dictionary from requested parameter names (or `__all__` if concat is
+            True) to :py:class:`xr.DataArray` s with the
+            requested parameter(s) as a column(s) and coordinates
+            formed by the dependencies.
+
+        Example:
+            Unpack a concatenated xr.Dataset with
+
+                xds, = ds.get_data_as_xarray(concat=True).values()
+        """
+        import xarray as xr
+        datadict = self.get_parameter_data(*params,
+                                           start=start,
+                                           end=end)
+
+        data_arrs = {}
+        for name, subdict in datadict.items():
+            index = self._generate_pandas_index(subdict)
+            arr = self._data_to_dataframe(subdict, index).to_xarray()[name]
+            arr.attrs["label"] = self.paramspecs[name].label
+            arr.attrs["unit"] = self.paramspecs[name].unit
```

Comment:
Why not include all of the paramspec elements, rather than just label and unit?

Reply:
Those were the minimum sweet of things I could imagine wanting. By all means, add more.